### DISCUSSION TITLE
method for (panel) -> (UPS) -> (CDU)

### DISCUSSION THREAD

**Discussion by ehopkins99** [opendcim/openDCIM#1492] [Q&A]
I'm documenting a small in-office data center and we frequently have rack mounted UPS (multiple) feeding additional racked PDUs and vertical power strips.  The openDCIM data model insists that CDUs must be fed from a power panel.  Any recommendations on how to depict:  

panel -> (power input) UPS (multiple power outputs) -> (power input) PDUs (multiple power outputs)

I can fudge the PDU information and add one additional power connection, but that seems wrong.  It feels like I'm saying "UPS output is connected to the extra PDU output".  

Ideally the CDU power source GUI and data would allow for powerSrcType=Panel or powerSrcType=otherCDU

Thanks for any suggestions

**Comment by samilliken** [comment]
Treat the UPS as a power panel.   Arbitrarily assign “breaker numbers” to your plugs.   Label them.

 

Scott

 

From: ehopkins99 
Sent: Thursday, January 11, 2024 11:09 AM
To: opendcim/openDCIM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [opendcim/openDCIM] method for (panel) -> (UPS) -> (CDU) (Discussion #1492)

 

I'm documenting a small in-office data center and we frequently have rack mounted UPS (multiple) feeding additional racked PDUs and vertical power strips. The openDCIM data model insists that CDUs must be fed from a power panel. Any recommendations on how to depict:

panel -> (power input) UPS (multiple power outputs) -> (power input) PDUs (multiple power outputs)

I can fudge the PDU information and add one additional power connection, but that seems wrong. It feels like I'm saying "UPS output is connected to the extra PDU output".

Ideally the CDU power source GUI and data would allow for powerSrcType=Panel or powerSrcType=otherCDU

Thanks for any suggestions

**Reply by LFrank2021** [reply]
Hi,
sorry for backpacking on this thread.
openDCIM currently supports 3 objects for power:
- CDU (the power strips inside the Rack)
- Power Panel (some kind of multi breaker, multi phase power distribution)
- Power Supply (Generator, grid, ...)

The CDU can be added through the rack management.
Power Panels can be added through the Power Management dialogue.
How can Power Supplies be added?
I can't seem to find the appropriate dialogue.

Kind regards
Frank

**Comment by ehopkins99** [comment]
Thanks for the quick response. Implementing that now

—Eric

**Comment by samilliken** [comment]
A power source is simply a power panel with no parent.

 

Scott

**Reply by LFrank2021** [reply]
Thank you. That is easy and genius.

Kind regards
Frank
